addon/controllers/fd-editform-constructor.js

```diff
@@ -45,17 +45,45 @@ export default Ember.Controller.extend({
   */
   selectedControl: undefined,
 
+  prevTab: undefined,
+
+  applicationController: Ember.inject.controller('application'),
+
+  configPanrlTabsWidth: Ember.computed.alias('applicationController.configPanrlTabsWidth'),
+
   actions: {
-    close() {
-      this.transitionToRoute('fd-appstruct-form');
-    },
 
-    toggleConfigPanel() {
-      if (this.toggleProperty('configPanelState')) {
-        this.transitionToRoute('fd-editform-constructor.form-config-panel');
-      } else {
-        this.transitionToRoute('fd-editform-constructor');
+    toggleConfigPanel(currentTab) {
+      let configPanelSidebar = Ember.$('.ui.sidebar.config-panel');
+      let configPanelSidebarVisible = configPanelSidebar.hasClass('visible');
+
+      if (this.prevTab === currentTab || this.prevTab === undefined || !configPanelSidebarVisible) {
+        let sidebar = Ember.$('.ui.sidebar.main.menu');
+
+        configPanelSidebar.sidebar({
```

Comment:
The sidebar is initialized in `ui-sidebar` component, `closable` and `dimPage` options can be passed [in the template](https://github.com/Flexberry/ember-flexberry-designer/pull/59/files#diff-dba549552b4c4a3061966944f9a713c8R32).